### PR TITLE
ARROW-10904: [Python][CI][Packaging] Add support for Python 3.9 macOS wheels

### PR DESCRIPTION
**Pull request by terencehonles**
Picking back up from #8386 in case I can handle this or someone else wants to work off of it.

### CODE REVIEW

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-osx-*-cp39

**Comment by terencehonles** [comment]
@kou / @kszucs you may already know but it looks like the crossbow submitter / github actions bot isn't working(?). It doesn't look like multibuild has tasks for GitHub Actions, but I wanted to see how far the Travis OS X commands would get.

**Comment by kszucs** [comment]
Yes, the comment bot seems to be failing https://github.com/apache/arrow/actions/runs/462228610
This is most likely caused by the recent security changes, so we need to update the [comment_bot.yml](https://github.com/apache/arrow/blob/master/.github/workflows/comment_bot.yml) action to use verified plugins only.

**Comment by terencehonles** [comment]
@kszucs I noticed there was a list of allowed actions. Could the allow list just be updated or is that enforced at the apache (org) level rather than the repo level?

**Comment by kszucs** [comment]
I'm afraid that it is enforced at organization level, so we need to update that action.

**Comment by kszucs** [comment]
Should be fixed by https://github.com/apache/arrow/pull/9103

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-*-cp39

**Comment by terencehonles** [comment]
> Should be fixed by #9103

Thanks @kszucs !

**Comment by terencehonles** [comment]
@github-actions crossbow submit wheel-osx-*-cp39

**Comment by terencehonles** [comment]
@github-actions crossbow submit -g conda -g wheel

**Comment by terencehonles** [comment]
@kou not sure if you might have a chance to review the PR, it looks like the change to GitHub actions seems to be working.

File: dev/tasks/python-wheels/github.osx.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          install_wheel arrow
+
+          # run unit tests before removing the system libraries (osx-build.sh)
+          ulimit -S -n 4096 && run_unit_tests arrow
+
+          # run the import tests (osx-build.sh)
+          run_import_tests
+
+          # restore the original Cellar
+          CELLAR=$(brew --cellar)
```
Comment (kou) [inline dev/tasks/python-wheels/github.osx.yml:95]: Can we remove this because we already have `CELLER=`?

**Review by kou — APPROVED** [review]
+1

Great!

File: dev/tasks/python-wheels/github.osx.yml (hunk 1 above)
Comment (terencehonles) [reply dev/tasks/python-wheels/github.osx.yml]: done, thanks

**Comment by kou** [comment]
Thanks!

**Comment by nealrichardson** [comment]
FWIW I have a mac with macOS 10.14 and I confirmed that I can install the high sierra wheel that this produced.

**Comment by terencehonles** [comment]
> FWIW I have a mac with macOS 10.14 and I confirmed that I can install the high sierra wheel that this produced.

Thanks, hopefully this made the cut for the next release and there are no issues with the build system changing :slightly_smiling_face:

**Comment by aberres** [comment]
Great work, looking forward to get this released!

> Thanks, hopefully this made the cut for the next release and there are no issues with the build system changing 🙂

Is there any place where one can find a release schedule?

**Comment by kszucs** [comment]
You can follow the release process on the development mailing list (https://arrow.apache.org/community/).

**Comment by mafrosis** [comment]
Glad to see this is coming. What date will the wheel be available?

**Comment by aberres** [comment]
> Glad to see this is coming. What date will the wheel be available?

An RC for 3.0 is on its way. See https://lists.apache.org/list.html?dev@arrow.apache.org

**Comment by mafrosis** [comment]
Ah thanks for that, but I haven't tested pyarrow 3 at all - I was hoping for a built wheel of 2.0.0 for Python 3.9 on macos

**Comment by aberres** [comment]
`Downloading pyarrow-3.0.0-cp39-cp39-macosx_10_13_x86_64.whl (14.2 MB)`

Yay, thanks!

**Pull request closed without merging**